analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -180,12 +173,35 @@ class GoMod(
             val parent = parsePackageEntry(columns[0])
             val child = parsePackageEntry(columns[1])
 
-            result += Edge(parent, child)
+            result.addEdge(parent, child)
         }
 
         return result
     }
 
+    /**
+     * Remove the packages from the given [graph] that are not used by the module by running the _go mod why_ command
+     * repeatedly in [projectDir].
+     */
+    private fun filterUsedPackages(graph: Graph, projectDir: File): Graph {
+        val usedPackageNames = mutableSetOf<String>()
+
+        graph.keys.chunked(WHY_CHUNK_SIZE).forEach { ids ->
+            val pkgNames = ids.map { it.name }.toTypedArray()
+            usedPackageNames +=
+                parseWhyOutput(run(projectDir, "mod", "why", *pkgNames).requireSuccess().stdout.lineSequence())
```

Comment:
https://golang.org/ref/mod#go-mod-why says that "The -vendor flag causes go mod why to ignore imports in tests of packages outside the main module (as go mod vendor does).". So, would it make sense to add the `-vendor´ flag?

Reply:
I decided against this because in the following it is stated that from go 1.16 or higher that flag does not make any difference. OTOH, it would probably do no harm to add it.